pkg/policy/k8s/cilium_network_policy.go

```diff
@@ -134,7 +134,9 @@ func (p *policyWatcher) resolveCiliumNetworkPolicyRefs(
 	translatedCNP := cnp.DeepCopy()
 
 	// Resolve ToService references
-	p.resolveToServices(key, translatedCNP)
+	if _, exists := p.toServicesPolicies[key]; exists {
+		p.resolveToServices(key, translatedCNP)
```

Comment:
I am happy about it from the scalability point of view, but I am not sure if it's correct. 
`toServicesPolicies` is updated in the parent function here: https://github.com/cilium/cilium/blob/9b7da39de54453b229a88c5ff5dbcf09037f1d5f/pkg/policy/k8s/cilium_network_policy.go#L77-L81

Imagine that someone removes `toServices` in CNP. Shouldn't we still perform `p.clearCNPForService(key, svcID)` for whatever reason it is there?

Reply:
very good catch!

Reply:
OK, that should do it.